azerothcore/azerothcore-wotlk
# fix(Core/Spells): Correct hunter's Lock and Load procs

- Move related code to spell scripts
- Fixes it proc'ing when the target is immune to snares
- Partial port of https://github.com/TrinityCore/TrinityCore/commit/746b838870e93d9843a089613ce7f1cd160b6ea5



## Changes Proposed:
- Move related code to spell scripts
- Fixes it proc'ing when the target is immune to snares
- Partial port of https://github.com/TrinityCore/TrinityCore/commit/746b838870e93d9843a089613ce7f1cd160b6ea5


## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8011

## SOURCE:


## Tests Performed:

- Tested using immolation traps against immune & non immune targets: no procs happened, as expected.
- Tested using frost & freezing traps against immune & non immune targets: no procs happened when the targets were immune, otherwise the spell proc'ed normally, as expected.


## How to Test the Changes:


1. Create a hunter and learn the Lock and Load talent from the Survival spec tree
2. .npc add temp 9938 (Magmus, boss, immune) / npc add temp 26829 (Frenzied Worgen, random creature, no immunities)
3. Use the trap spells and lure them individually into the trap, see if the spell procs or not 

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
